lnpbp-0021.md

```diff
@@ -104,57 +101,62 @@ data AttachmentType ::
     description [Ascii ^ 1..20]
 
 interface RGB21
-    global name :: RGBTypes.AssetNaming
+    -- Asset specification containing ticker, name, precision etc.
+    global spec :: RGBTypes.DivisibleAssetSpec
     
     -- Contract text is separated from the name since it must not be
     -- changeable by the issuer.
     global terms :: RGBTypes.RicardianContract
 
     -- Data for all issued tokens
-    global tokensData+ :: TokenData
-    global engravingData* :: EngravingData
-    global isFractional :: Bool
-    global attachmentTypes+ :: AttachmentType
+    global tokens* :: TokenData
+    global engravings* :: EngravingData
+    global attachmentTypes* :: AttachmentType
 
     -- Right to do a secondary (post-genesis) issue
     owned inflationAllowance* :: RGBTypes.Amount
     -- Right to update asset name
     owned updateRight?
 
     -- Ownership right over assets
-    owned assetOwner+ :: Allocation
+    owned assetOwner* :: Allocation
 
     genesis       -> name
                    , terms,
                    , reserves PoR*
-                   , tokens tokenData+
-                   , assetOwner+
+                   , tokens*
+                   , assetOwner*
```

Comment:
Genesis now can just produce an issue right and not tokens (i.e. they may be issued later). Thus also a global type change and `Issue` parameters. This makes RGB21 compatible with RGB20